### ISSUE TITLE
Feature Request: Support key finders

### ISSUE THREAD

**Issue by IzzySoft** [Freeyourgadget/Gadgetbridge#774]
#### Before opening an issue please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
A new feature suggestion

#### Your wearable device is:
Pebble Time Steel

#### Your android version is:
5.1

#### Your Gadgetbridge version is:
0.19.4

----

There are several "key finders" available e.g. at Amazon, using BLE. They're usually 2-way (with a button), and the price range starts at [about EUR 5](https://www.amazon.de/dp/B01M2WF51S "Bluetooth anti-loss Tracker with carabiner") (unrated), having a good candidate at [about EUR 10](https://www.amazon.de/dp/B0734PWCYT "GOMAN Key Finder") (5 stars at 16 votes), another interesting one [at EUR 10](https://www.amazon.de/dp/B07259HV33 "Anti Lost Smart Key Finder belt-clip") (unrated), Two-for-One [at ~EUR 15](https://www.amazon.de/dp/B071FB1F7C "XinXu Key Finder") (5 stars at 12 votes) – and of course no definite "upper limit". All of them need a companion app, but interestingly multiple of them point to the very same ([iSearching](https://play.google.com/store/apps/details?id=com.lenzetech.antilost), with a lousy rating of 2.0 at 222 votes – though I bet [iKeeper_pro](https://play.google.com/store/apps/details?id=com.ds.ikeeper), [Smart Button nurum](https://play.google.com/store/apps/details?id=com.dothing.nurum) and others could be used as well) – which indicates a "common protocol". Behavior is quite obvious:

**In reach and connected:**

* press the button on the finder › phone rings an alarm
* press a button in the app › finder makes a noise
* use the finder as remote shutter for the camera, or to start a voice recording

**Connection is lost:**

* phone alarm rings

**Out of reach:**

* app shows "last seen time"
* for some, app remembers "last seen location" and can show it on a map

Now, where should Gadgetbridge come in? Of course as replacement for the proprietary companion. The "finder" could be dealt with like a "basic gadget": in the overview, next to the name would only be "connected/disconnected", a button to make it ring and, if possible, the battery level. Obviously we have no statistics to show (except if you count connection/absence times, at what time you "pushed some button", and where it was located at what time). All above behavior (maybe except for camera shutter/voice recorder, but even that would be doable) could be implemented (though location would be "not exact"). For "connection lost", it would be great to have the wearable (Pebble, Mi:Band, …) vibrate instead of / in addition to the phone alert.

----

I know, a current show-stopper is that GB can only handle 1 active device at the moment – and for the feature described, 2 devices needed to be active simultaneously. Thus I'm completely aware of the fact this feature cannot be expected "tomorrow". **Until then,** as an probably easier-to-implement **work-around:** could there be **an intent provided for e.g. *Tasker* to tell GB it should send a notification** with a given text (and optionally, of a given "category" – such as "info", "alert") to the device connected? To protect that against "abusing apps", there could be some whitelist/blacklist setting (and upon "first contact" user can tell where it belongs, for easy setup, while being able to change that later).

**Comment by danielegobbetti** [comment]
Also this one looks like an interesting device / use case: https://chipolo.net/pages/card

**Comment by IzzySoft** [comment]
@danielegobbetti yes. But not with its own app. Seeing how many trackers are on the product page (I didn't even see any pictures because I was unwilling to permit Javascript from X 3rd party sources), I don't trust it concerning privacy. But that was exactly my point for GB :smile:

Btw: If one knows a very small candidate (smaller than 20mm) …

**Comment by TaaviE** [comment]
I ordered a bunch of different models I saw on eBay/Aliexpress. There's a really informative issue about these tags here https://github.com/sylvek/itracing2/issues/5.

**Comment by TaaviE** [comment]
I've been trying to write a value to my iTags, but for some reason when running:
```
        getQueue().clear();
        TransactionBuilder builder = new TransactionBuilder("beeping");
        builder.write(getCharacteristic(UUID.fromString("00002a06-0000-1000-8000-00805f9b34fb")), new byte[]{0x02});
        builder.queue(getQueue());
```
the tag doesn't beep, but when writing the same value with nRF Connect app the tag starts beeping, Both the UUID and the value are copied from the nRF Connect app, so they should be right. Logs of both apps indicate the same write 
```
no.nordicsemi.android.mcp D/BluetoothGatt: writeCharacteristic() - uuid: 00002a06-0000-1000-8000-00805f9b34fb
no.nordicsemi.android.mcp D/BluetoothGatt: onCharacteristicWrite() - Device=[CENSORED] UUID=00002a06-0000-1000-8000-00805f9b34fb Status=0
```

and

```
nodomain.freeyourgadget.gadgetbridge.debug D/BluetoothGatt: writeCharacteristic() - uuid: 00002a06-0000-1000-8000-00805f9b34fb
nodomain.freeyourgadget.gadgetbridge.debug D/BluetoothGatt: onCharacteristicWrite() - Device=[CENSORED] UUID=00002a06-0000-1000-8000-00805f9b34fb Status=0
```

What am I doing wrong?

**Comment by cpfeiffer** [comment]
There's nothing obviously wrong (that's why I couldn't suggest something quickly).

Create a btsnoop of both ways and see where the difference is.

Be sure to use createTransactioBuilder(), so that your *Support class will be attached to the queue as a listener.
